src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -189,6 +197,106 @@ def add_storage(
+
+            # We build the FieldType for this temporary array.
+            dims: list[Dimension] = []
+            for dim, _ in tmp_domain:
+                dims.append(
+                    Dimension(
+                        value=dim,
+                        kind=(
+                            DimensionKind.VERTICAL
+                            if self.column_axis is not None and self.column_axis.value == dim
+                            else DimensionKind.HORIZONTAL
+                        ),
+                    )
+                )
+            assert isinstance(tmp.dtype, str)
+            type_ = ts.FieldType(dims=dims, dtype=as_scalar_type(tmp.dtype))
+            self.storage_types[tmp_name] = type_
+
+            # N.B.: skip generation of symbolic strides and just let dace assign default strides, for now.
+            # Another option, in the future, is to use symbolic strides and apply auto-tuning or some heuristics
+            # to assign optimal stride values.
+            tmp_shape, _ = new_array_symbols(tmp_name, len(dims))
+            tmp_offset = [
+                dace.symbol(unique_name(f"{tmp_name}_offset{i}")) for i in range(len(dims))
+            ]
+            _, tmp_array = program_sdfg.add_array(
+                tmp_name, tmp_shape, as_dace_type(type_.dtype), offset=tmp_offset, transient=True
+            )
+
+            # Loop through all dimensions to visit the symbolic expressions for array shape and offset
+            # This loop will produce a map of symbol assignments, where the origin value is the result of a tasklet.
+            for (_, (begin, end)), offset_sym, shape_sym in zip(
+                tmp_domain,
+                tmp_array.offset,
+                tmp_array.shape,
+            ):
+                offset_tasklet = defs_state.add_tasklet(
+                    "offset",
+                    code=f"__result = - {begin.value}",
+                    inputs={},
+                    outputs={"__result"},
+                )
+                offset_var = unique_var_name()
+                program_sdfg.add_scalar(offset_var, symbol_dtype, transient=True)
+                offset_node = defs_state.add_access(offset_var)
+                defs_state.add_edge(
+                    offset_tasklet,
+                    "__result",
+                    offset_node,
+                    None,
+                    dace.Memlet.simple(offset_var, "0"),
+                )
+
+                shape_tasklet = defs_state.add_tasklet(
+                    "shape",
+                    code=f"__result = {end.value} - {begin.value}",
+                    inputs={},
+                    outputs={"__result"},
+                )
+                shape_var = unique_var_name()
+                program_sdfg.add_scalar(shape_var, symbol_dtype, transient=True)
+                shape_node = defs_state.add_access(shape_var)
+                defs_state.add_edge(
```

Comment:
According to my understanding, you should be able to put even complex expressions in assignments on interstate edge.